code/main/media/mediaModule.brs

```diff
@@ -0,0 +1,245 @@
+                            mediaEventType = mediaRequest.xdmData.xdm.eventType
+                            m._handleMediaEvent(mediaEventType, mediaRequest.requestId, sessionId, location, mediaRequest.xdmData, mediaRequest.tsObject)
+                        end for
+                        m._kickRequestQueue()
+                    end if
+                end if
+            end for
+        end sub,
+
+        _handleMediaEvent: sub(mediaEventType as string, requestId as string, sessionId as string, location as string, xdmData as object, tsObject as object)
+            path = _adb_EdgePathForEventType(mediaEventType, location, m._CONSTANTS.MEDIA.EVENT_TYPES)
+            if not _adb_isEmptyOrInvalidString(path)
+                xdmData.xdm["_id"] = _adb_generate_UUID()
+                xdmData.xdm["timestamp"] = tsObject.ts_inISO8601
+                xdmData.xdm["mediaCollection"]["sessionID"] = sessionId
+
+                meta = {}
+                ' TODO: sanitize the xdmData object before sending to the backend.
+                m._edgeRequestWorker.queue(requestId, [xdmData], tsObject.ts_inMillis, meta, path)
+                m._kickRequestQueue()
+            else
+                _adb_logError("_handleMediaEvent() - mediaEventName is invalid: " + mediaEventType)
+            end if
+        end sub,
+
+        _getEdgeConfig: function() as object
+            configId = m._configurationModule.getConfigId()
+            if _adb_isEmptyOrInvalidString(configId)
+                return invalid
+            end if
+            ecid = m._identityModule.getECID()
+            if _adb_isEmptyOrInvalidString(ecid)
+                return invalid
+            end if
+            return {
+                configId: configId,
+                ecid: ecid,
+                edgeDomain: m._configurationModule.getEdgeDomain()
+            }
+        end function,
+
+        _processQueuedRequests: function() as dynamic
+            responseEvents = []
+
+            if not m._edgeRequestWorker.hasQueuedEvent()
+                ' no requests to process
+                return responseEvents
+            end if
+
+            edgeConfig = m._getEdgeConfig()
+            if edgeConfig = invalid
+                _adb_logVerbose("_processQueuedRequests() - Cannot send network request, invalid configuration.")
+                return responseEvents
+            end if
+
+            responses = m._edgeRequestWorker.processRequests(edgeConfig.configId, edgeConfig.ecid, edgeConfig.edgeDomain)
+
+            return responses
+        end function,
+
+        dump: function() as object
+            ' TODO: update the dump info when adding the integration tests
+            return {
+                requestQueue: m._edgeRequestWorker._queue
+            }
+        end function
+    })
+    return module
+end function
+
+function _adb_EdgePathForEventType(mediaEventType as string, location as string, supportedTypes as object) as dynamic
```

Comment:
`_adb_getEdgePathForMediaEvent(type as string, supportedTypes as object)`

We don't need location. Location should be handled by the edge module when we support location hints.

Reply:
If we validate media events in the API layer we dont need to validate with supportedTypes here. I would prefer entry level filtering rather than checking it everywhere.